tests/evaluation/metrics_test.py

```diff
@@ -0,0 +1,134 @@
+"""Tests for GtsfmMetric and GtsfmMetricsGroup. 
+
+Authors: Akshay Krishnan
+"""
+import copy
+import numpy as np
+import unittest
+import tempfile
+import os
+
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+
+class TestGtsfmMetric(unittest.TestCase):
+    """Tests for the Metrics class."""
+
+    def setUp(self):
+        super().setUp()
+        self._metric_dict_quartiles = {
+            "foo_metric": {
+                "summary": {
+                    "min": 0,
+                    "max": 7,
+                    "median": 3.5,
+                    "mean": 3.5,
+                    "stddev": 2,
+                    "quartiles": {
+                        "q0": 0,
+                        "q1": 1,
+                        "q2": 3,
+                        "q3": 5,
+                        "q4": 7,
+                    },
+                },
+                "full_data": [0, 1, 2, 3, 4, 5, 6, 7],
+            }
+        }
+        self._metric_dict_histogram = {
+            "bar_metric": {
+                "summary": {
+                    "min": 0,
+                    "max": 3,
+                    "median": 1.5,
+                    "mean": 1.5,
+                    "stddev": 2,
+                    "histogram": {
+                        "0": 2,
+                        "1": 2,
+                        "2": 2,
+                        "3": 2,
+                    },
+                },
+                "full_data": [0, 1, 0, 2, 2, 1, 3, 3],
+            }
+        }
+        self._metric_dict_no_data = copy.deepcopy(self._metric_dict_histogram)
+        del self._metric_dict_no_data["bar_metric"]["full_data"]
+
+    def test_create_scalar_metric(self):
+        metric = GtsfmMetric("a_scalar", 2)
+        self.assertEqual(metric.name, "a_scalar")
+        np.testing.assert_equal(metric.data, np.array([2]))
+        self.assertEqual(metric.plot_type, GtsfmMetric.PlotType.BAR)
+
+    def test_create_1d_distribution_metric(self):
```

Comment:
Could you explain in a comment why box is the only valid Plot type here?

Reply:
Its not the only valid one, but the default one.

Reply:
explained this further in the constructor description.